File: spec/rust/tests/test_type_ternary_2nd_falsy.rs
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::type_ternary_2nd_falsy::*;


#[test]
fn test_type_ternary_2nd_falsy() {
    let bytes = fs::read("../../src/switch_integers.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let res = TypeTernary2ndFalsy::read_into(&_io, None, None);
    let r : OptRc<TypeTernary2ndFalsy>;

    if let Err(err) = res {
        panic!("{:?}", err);
    } else {
        r = res.unwrap();
    }

    assert_eq!(*r.v_false().unwrap(), false);
    assert_eq!(*r.v_int_zero().unwrap(), 0);
    assert_eq!(*r.v_int_neg_zero().unwrap(), 0);
    assert_eq!(*r.v_float_zero().unwrap(), 0.0);
    assert_eq!(*r.v_float_neg_zero().unwrap(), -0.0);
    assert_eq!(*r.v_str_w_zero().unwrap(), "0");
    assert_eq!(r.v_str_w_zero().unwrap().len(), 1);
    assert_eq!(*r.ut().m(), 7);
    assert_eq!(r.v_null_ut().unwrap().is_none(), true);
    // v_null_ut() always uninited
    // assert_eq!(*r.v_null_ut().unwrap().m(), 0);
    assert_eq!(*r.v_str_empty().unwrap(), "");
    assert_eq!(r.v_str_empty().unwrap().len(), 0);
    assert_eq!(r.int_array().len(), 2);
    assert_eq!(r.v_int_array_empty().unwrap().len(), 0);
}

File: spec/javascript/test_type_ternary_2nd_falsy.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('TypeTernary2ndFalsy', 'src/switch_integers.bin', function(r, TypeTernary2ndFalsy_) {
  assert.strictEqual(r.vFalse, false);
  assert.strictEqual(r.vIntZero, 0);
  assert.strictEqual(r.vIntNegZero, 0);
  assert.strictEqual(r.vFloatZero, 0.0);
  assert.strictEqual(r.vFloatNegZero, -0.0);
  assert.strictEqual(r.vStrWZero, "0");
  assert.strictEqual(r.vStrWZero.length, 1);
  assert.strictEqual(r.ut.m, 7);
  assert.strictEqual(r.vNullUt, undefined);
  assert.strictEqual(r.vStrEmpty, "");
  assert.strictEqual(r.vStrEmpty.length, 0);
  assert.strictEqual(r.intArray.length, 2);
  assert.strictEqual(r.vIntArrayEmpty.length, 0);
});
